### PR TITLE
Convert `Host` class to JS

### PR DESCRIPTION
**Pull request by robertknight**
This is a straightforward conversion with decaffeinate, followed by
manual fixups of lint errors etc.

### CODE REVIEW

File: src/annotator/host.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      }
+    }
+
+    // Make a copy of all config settings except `config.sidebarAppUrl`, the app base URL,
+    // and `config.pluginClasses`
+    const configParam =
+      'config=' +
+      encodeURIComponent(
+        JSON.stringify(
+          Object.assign({}, config, {
```
Comment (LMS007) [inline src/annotator/host.js:38]: might be able to refactor this a tad more. Is there any reason to initialize sidebarAppUrl or pluginClasses to undefined? seems like that could just be omitted.

either `JSON.stringify(config)` or `JSON.stringify({...config})` if copy is needed

File: src/annotator/host.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    let externalContainer = null;
+
+    if (config.externalContainerSelector) {
+      // Use the native method to also validate the input
+      externalContainer = document.querySelector(
+        config.externalContainerSelector
+      );
+    }
+    super(...arguments);
```
Comment (LMS007) [inline src/annotator/host.js:65]: any reason this can't be the first statement in the ctor?

File: src/annotator/test/host-test.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    );
+    if (!element) {
+      element = document.createElement('div');
+    }
+    return new Host(element, config);
+  };
+
+  beforeEach(() => {
+    // Disable any Host logging.
+    sandbox.stub(console, 'log');
```
Comment (LMS007) [inline src/annotator/test/host-test.js:24]: Do you have a preference between sandbox or

`sinon.stub(console, 'log');`
`console.log.restore()` 

Just wondering when to use one or the other

File: src/annotator/test/host-test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        {
+          $highlight: true,
+        },
+      ]);
+      assert.notCalled(frame.contentWindow.focus);
+    });
+  });
+
+  describe('config', () => {
+    it('disables highlighting if showHighlights: false is given', done => {
```
Comment (LMS007) [inline src/annotator/test/host-test.js:91]: I don't see showHighlights condition anywhere in this PR

**Review by LMS007 — APPROVED** [review]
Just a few tiny suggestions / questions, then good to go!
👍

File: src/annotator/host.js (hunk 1 above)
Comment (robertknight) [reply src/annotator/host.js:38]: The comment above this line explains that these parameters should not be included in the config passed in the sidebar's fragment URL. I'll rewrite this to make it clearer.

File: src/annotator/host.js (hunk 2 above)
Comment (robertknight) [reply src/annotator/host.js:65]: I can't see any problem with doing so. I'll make this change.

File: src/annotator/test/host-test.js (hunk 3 above)
Comment (robertknight) [reply src/annotator/test/host-test.js:24]: If you're only stubbing one or two functions, `sinon.stub(...)` is simpler. If you're stubbing a lot of methods, a sandbox provides a convenient way to undo all the stubs at once IIRC. If you find yourself stubbing a lot of methods though, you might want to consider an alternative approach such as injecting a fake version of the object that has the stubbed methods (eg. like we do with `window` in a few places)

File: src/annotator/test/host-test.js (hunk 4 above)
Comment (robertknight) [reply src/annotator/test/host-test.js:91]: The comment refers to passing `showHighlights: false` in the `createHost(...)` call on the line below.

File: src/annotator/host.js (hunk 2 above)
Comment (robertknight) [reply src/annotator/host.js]: Update: There _is_ a reason that `super(...)` needs to be called part-way down the function. It turns out that the `super(...)` call results in the `BucketBar` constructor being called and that class in turn assumes that it is called after the bucket bar's container element (`.annotator-frame`) is in the DOM. This needs refactoring, but I'll do that as a separate task.

**Comment by robertknight** [comment]
I've revised a couple of things in response to comments. It turns out that having the `super(...)` call part-way down the constructor is currently needed for non-obvious reasons that will need more substantial refactoring to fix. I added a comment to explain why.

**Pull request merged**